docs/sql-ref-syntax-qry-select-case.md

```diff
@@ -0,0 +1,114 @@
+
+### Syntax
+
+```sql
+CASE [ expression ] { WHEN boolean_expression THEN then_expression } [ ... ]
+    [ ELSE else_expression ]
+END
+```
+
+### Parameters
+    
+* **WHEN**
+
+    Specific a boolean condition ,under which to return the `THEN` result, `WHEN` must exist in `CASE` clause.
+    
+* **THEN**
+
+    Specific a result base the `WHEN` condition, `THEN` must exist in `CASE` clause.
+    
+* **ELSE**
+
+    Specific a default result for the `CASE` rules, it is optional, if user don't use else then the `CASE` will not have default result.
+    
+* **END**
+
+    Key words to finish a case clause, `END` must exist in `CASE` clause.
+    
+* **boolean_expression**
+
+    Specific specified condition, it should be boolean type.
+    
+* **then_expression**
+
+    Specific the then expression based on the `boolean_expression` condition, `then_expression` and `else_expression` should all be same type or coercible to a common type.
+    
+* **else_expression**
+
+    Specific the default expression, `then_expression` and `else_expression` should all be same type or coercible to a common type.
+    
+### Examples
+
+```sql
+CREATE TABLE person (id INT, name STRING, age INT);
+INSERT INTO person VALUES
+    (100, 'John', 30),
+    (200, 'Mary', NULL),
+    (300, 'Mike', 80),
+    (400, 'Dan', 50);
+
+SELECT id, CASE WHEN id > 200 THEN 'bigger' ELSE 'small' END FROM person;
++------+--------------------------------------------------+
+|  id  | CASE WHEN (id > 200) THEN bigger ELSE small END  |
++------+--------------------------------------------------+
+| 100  | small                                            |
+| 200  | small                                            |
+| 300  | bigger                                           |
+| 400  | bigger                                           |
++------+--------------------------------------------------+
+
+SELECT id, CASE id WHEN 100 then 'bigger' WHEN  id > 300 THEN '300' ELSE 'small' END FROM person;
++------+-----------------------------------------------------------------------------------------------+
+|  id  | CASE WHEN (id = 100) THEN bigger WHEN (id = CAST((id > 300) AS INT)) THEN 300 ELSE small END  |
++------+-----------------------------------------------------------------------------------------------+
+| 100  | bigger                                                                                        |
+| 200  | small                                                                                         |
+| 300  | small                                                                                         |
+| 400  | small                                                                                         |
++------+-----------------------------------------------------------------------------------------------+
+
+SELECT * FROM person where CASE 1 = 1 WHEN 100 THEN 'big' WHEN 200 THEN 'bigger' WHEN 300 THEN 'biggest' ELSE 'small' END = 'small';
```

Comment:
`where` -> `WHERE` and how about this?
```
SELECT * FROM person
    WHERE
        CASE 1 = 1
            WHEN 100 THEN 'big'
            WHEN 200 THEN 'bigger'
            WHEN 300 THEN 'biggest'
            ELSE 'small'
        END = 'small';
```

Reply:
Why did you use `1 = 1` here?

Reply:
It is case for `CASE` expression, and we have `CASE` column case above.

Reply:
It is case for `CASE` expression, and we have `CASE` column case above.